lib/ecmascript.ts

```diff
@@ -817,16 +514,366 @@ export const ES = ObjectAssign({}, ES2020, {
+}
+
+export function ToTemporalRoundingIncrement(options, dividend, inclusive) {
+  let maximum = Infinity;
+  if (dividend !== undefined) maximum = dividend;
+  if (!inclusive && dividend !== undefined) maximum = dividend > 1 ? dividend - 1 : 1;
+  const increment = GetNumberOption(options, 'roundingIncrement', 1, maximum, 1);
+  if (dividend !== undefined && dividend % increment !== 0) {
+    throw new RangeError(`Rounding increment must divide evenly into ${dividend}`);
+  }
+  return increment;
+}
+
+export function ToTemporalDateTimeRoundingIncrement(options, smallestUnit) {
+  const maximumIncrements = {
+    year: undefined,
+    month: undefined,
+    week: undefined,
+    day: undefined,
+    hour: 24,
+    minute: 60,
+    second: 60,
+    millisecond: 1000,
+    microsecond: 1000,
+    nanosecond: 1000
+  };
+  return ToTemporalRoundingIncrement(options, maximumIncrements[smallestUnit], false);
+}
+
+export function ToSecondsStringPrecision(options): {
+  precision: 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 'auto' | 'minute';
+  unit: string;
+  increment: number;
+} {
+  const smallestUnit = ToSmallestTemporalUnit(options, undefined, ['year', 'month', 'week', 'day', 'hour']);
+  switch (smallestUnit) {
+    case 'minute':
+      return { precision: 'minute', unit: 'minute', increment: 1 };
+    case 'second':
+      return { precision: 0, unit: 'second', increment: 1 };
+    case 'millisecond':
+      return { precision: 3, unit: 'millisecond', increment: 1 };
+    case 'microsecond':
+      return { precision: 6, unit: 'microsecond', increment: 1 };
+    case 'nanosecond':
+      return { precision: 9, unit: 'nanosecond', increment: 1 };
+    default: // fall through if option not given
+  }
+  let digits = options.fractionalSecondDigits;
+  if (digits === undefined) digits = 'auto';
+  if (Type(digits) !== 'Number') {
+    digits = ToString(digits);
+    if (digits === 'auto') return { precision: 'auto', unit: 'nanosecond', increment: 1 };
+    throw new RangeError(`fractionalSecondDigits must be 'auto' or 0 through 9, not ${digits}`);
+  }
+  if (NumberIsNaN(digits) || digits < 0 || digits > 9) {
+    throw new RangeError(`fractionalSecondDigits must be 'auto' or 0 through 9, not ${digits}`);
+  }
+  const precision = MathFloor(digits);
+  switch (precision) {
+    case 0:
+      return { precision, unit: 'second', increment: 1 };
+    case 1:
+    case 2:
+    case 3:
+      return { precision, unit: 'millisecond', increment: 10 ** (3 - precision) };
+    case 4:
+    case 5:
+    case 6:
+      return { precision, unit: 'microsecond', increment: 10 ** (6 - precision) };
+    case 7:
+    case 8:
+    case 9:
+      return { precision, unit: 'nanosecond', increment: 10 ** (9 - precision) };
+    default:
+      throw new RangeError(`fractionalSecondDigits must be 'auto' or 0 through 9, not ${digits}`);
+  }
+}
+
+export function ToLargestTemporalUnit(options, fallback, disallowedStrings = [], autoValue?: any) {
```

Comment:
Any is the current type, but it's definitely not the actual type :) As you mentioned elsewhere, I'll send a separate PR to specify these as actual types.

The only reason these changes are needed is so we can call this function without passing either the `disallowedStrings` or `autoValue` parameters - otherwise, tsc complains that we are missing parameters in the various calls.